test/helpers/test_helpers.cpp

```diff
@@ -146,6 +146,12 @@ unique_ptr<DBConfig> GetTestConfig() {
 	result->options.checkpoint_wal_size = 0;
 #else
 	result->options.checkpoint_on_shutdown = false;
+#endif
+#if defined(__linux__)
+	// Currently this is only supported on linux.
+	// Because we test this mode on linux, and the no-trim mode on other platforms,
+	// we get coverage with the option both on and off.
+	result->options.trim_free_blocks = true;
```

Comment:
I am not sure about this part. I think it's great to test this thoroughly, but it looks this also means we will stop testing duckdb on Linux with its default setting, and I can see potentially some problems being visible only with one of the two modes.

Unsure what's the way forward.
Maybe leaving default to false but having some unittester options to turn this on based on environment variable?

And also, I would consider this to not be __linux__ only, and when this is turned on all filesystems handle this properly (via Trim being a no-op).

Reply:
Thank you. I wasn't sure what to do here and was hoping for feedback.

Reply:
I can make that change. Although DUCKDB_ALTERNATIVE_VERIFY also tests a slightly different checkpoint/shutdown behavior, which could conceivably interact?

Reply:
Yes that is true, maybe putting it behind `DESTROY_UNPINNED_BLOCKS` would be better? Or `RUN_SLOW_VERIFIERS`?

Reply:
Updated to use DUCKDB_RUN_SLOW_VERIFIERS